### PR TITLE
fix(rules): include_subdirs qualified errors

### PR DESCRIPTION
**Pull request by rgrinberg**
(include_subdirs qualified) is forbidden for libraries with [stdlib]

Signed-off-by: Rudi Grinberg <me@rgrinberg.com>

### CODE REVIEW

File: src/dune_rules/ml_sources.ml [hunk 1]
```diff
@@ -413,7 +435,8 @@ let modules_of_stanzas dune_file ~dir ~scope ~lookup_vlib ~modules =
   >>| filter_partition_map
 
 let make dune_file ~dir ~scope ~lib_config ~loc ~lookup_vlib
-    ~include_subdirs:(_loc, (include_subdirs : Dune_file.Include_subdirs.t))
+    ~include_subdirs:
```
Comment (anmonteiro) [inline src/dune_rules/ml_sources.ml:438]: is this pattern destructured for readability? we could just pass `~include_subdirs` as we get it

**Review by anmonteiro — COMMENTED** [review]
could you add a test in e.g. `test/blackbox-tests/test-cases/stdlib-compilation.t` for this case?

File: src/dune_rules/ml_sources.ml [hunk 2]
```diff
@@ -315,13 +317,33 @@ let make_lib_modules ~dir ~libs ~lookup_vlib ~(lib : Library.t) ~modules =
         (Option.value ~default:Ordered_set_lang.standard lib.private_modules)
       ~src_dir:dir modules_settings
   in
-  let stdlib = lib.stdlib in
+  let () =
+    match (lib.stdlib, include_subdirs) with
+    | Some stdlib, Include Qualified ->
+      let main_message =
+        Pp.text "an stdlib library may not use (include_subdirs qualified)"
```
Comment (anmonteiro) [inline src/dune_rules/ml_sources.ml:324]:
```suggestion
        Pp.text "a library with (stdlib ...) may not use (include_subdirs qualified)"
```

File: src/dune_rules/ml_sources.ml (hunk 1 above)
Comment (rgrinberg) [reply src/dune_rules/ml_sources.ml:438]: We're also matching on `include_subdirs` in this function.

Comment (anmonteiro) [reply src/dune_rules/ml_sources.ml:438]: indeed, should have expanded more

**Comment by rgrinberg** [comment]
Added a test.

It might be worth it to experiment with melange using this extension to build the stdlib. Melange would benefit from not having its own custom rules and dune would benefit from some real world testing of this feature.

File: test/blackbox-tests/test-cases/include-qualified/stdlib.t [hunk 3]
```diff
@@ -0,0 +1,23 @@
+Libraries designed as the "stdlib" may not use (include_subdirs qualified)
+
+Although it's just an aritificial limitation and when we can lift it if we want
```
Comment (anmonteiro) [inline test/blackbox-tests/test-cases/include-qualified/stdlib.t:3]: typo

```suggestion
Although it's just an artificial limitation and when we can lift it if we want
```

**Review by anmonteiro — APPROVED** [review]
Thanks for adding the test

**Pull request merged**